tests/scripts/run-xunit-perf.py

```diff
@@ -0,0 +1,454 @@
+    build_perfharness(coreclrRepo, sandboxDir, extension, dotnetEnv)
+
+    my_env = setup_optimization_level(my_env, optLevel)
+
+    if not 'XUNIT_PERFORMANCE_MAX_ITERATION' in my_env:
+        my_env['XUNIT_PERFORMANCE_MAX_ITERATION'] = '21'
+    if not 'XUNIT_PERFORMANCE_MAX_ITERATION_INNER_SPECIFIED' in my_env:
+        my_env['XUNIT_PERFORMANCE_MAX_ITERATION_INNER_SPECIFIED'] = '21'
+
+    # Copy core overlay contents to sandbox
+    copytree(coreclrOverlay, sandboxDir)
+
+    # Determine benchmarks we will be running
+    benchmarks = []
+    if not sliceNumber == -1:
+        with open(sliceConfigFile) as jsonData:
+            data = json.load(jsonData)
+            if sliceNumber >= len(data["slices"]):
+                raise ValueError('Invalid slice number. %s is greater than the max number of slices %s' % (sliceNumber, len(data["slices"])))
+            for benchmark in data["slices"][sliceNumber]["folders"]:
+                benchmarks += [benchmark]
+
+    else:
+    # If slice was not specified, run everything in the coreclrPerf directory. Set benchmarks to an empty string
+        benchmarks = [{ 'directory' : '', 'extraFlags': ''}]
+
+    testFileExt = 'dll' if isLibrary else 'exe'
+
+    # Run benchmarks
+    failures = 0
+    lvRunId = 'Perf-%s' % etwCollection
+
+    for benchmark in benchmarks:
+        testFileExt = 'dll' if benchmark["extraFlags"] == '-library' else 'exe'
+        benchmarkDir = os.path.normpath(benchmark["directory"])
+
+        testPath = os.path.join(coreclrPerf, benchmarkDir)
+
+        for root, dirs, files in os.walk(testPath):
+            for f in files:
+                if f.endswith(testFileExt):
+                    benchname, ext = os.path.splitext(f)
+
+                    benchmarkOutputDir = os.path.join(sandboxOutputDir, 'Scenarios') if isScenarioTest else os.path.join(sandboxOutputDir, 'Microbenchmarks')
+                    benchmarkOutputDir = os.path.join(benchmarkOutputDir, etwCollection, benchname)
+
+                    failures += run_benchmark(benchname, root, my_env, sandboxDir, benchmarkOutputDir, testFileExt, stabilityPrefix, collectionFlags, lvRunId, etwCollection, isScenarioTest, arch, extension, executable)
+                    if not benchviewPath is None:
+                        generate_results_for_benchview(python, lvRunId, benchname, isScenarioTest, better, hasWarmupRun, benchmarkOutputDir, benchviewPath)
+
+    # Setup variables for uploading to benchview
+    pgoOptimized = 'pgo' if isPgoOptimized else 'nopgo'
+
+    # Upload to benchview
+    if not benchviewPath is None:
+        upload_to_benchview(python, coreclrRepo, benchviewPath, uploadToBenchview, benchviewGroup, runType, configuration, operatingSystem, etwCollection, optLevel, jitName, pgoOptimized, arch)
+
+    if not failures == 0:
+        log('%s benchmarks have failed' % failures)
+
+    return failures
+
+if __name__ == "__main__":
+    Args = parser.parse_args(sys.argv[1:])
+    sys.exit(main(Args))
```

Comment:
You should be able to call the built-in `exit` function instead.

Reply:
Python generally doesn't like the use of exit inside of scripts, stylistically

Reply:
https://stackoverflow.com/questions/19747371/python-exit-commands-why-so-many-and-when-should-each-be-used